src/Functions/Public/Connect-vRAServer.ps1

```diff
@@ -221,16 +214,49 @@
 
                 }
 
-                $Response = Invoke-RestMethod @Params
+                # Here we try with the first set of params, if it fails, it may be due to another configuration
+                # so we try again with the alternative option if available
+                try {
+                    $Response = Invoke-RestMethod @Params
+                } catch [Exception]{
+                    if ($_.Exception.Response.StatusCode -eq "BadRequest" -and $null -ne $AlternateJson) {
+                        $Params = @{
```

Comment:
I think the only thing you need to overwrite here is $Body... the Params splat is already in memory and can be reused from above.

Reply:
Will make the change

Reply:
Did this now, please take a look